students/jsward/session02/series.py

```diff
@@ -0,0 +1,79 @@
+
+
+def fibonacci(nums_to_compute):
+    # First solution
+    """
+    Print the Fibronacci number series.
+    :param nums_to_compute:
+    :return: none
+    """
+    results = [0, 1]
+    while len(results) <= nums_to_compute:
+        results.append(sum(results[-2:]))
+    return results[nums_to_compute]
+
+
+def fib(n):
+    # Solution from class
+    a, b = 0, 1
+    for _ in range(n - 1):
+        a, b = b, a + b
+    return b
+
+
+def lucas(n):
+    """
+    Print the Lucas number series
+    :param n: number from the Lucas seies to return:
+    :return: nth number from the Lucas series
+    """
+    results = [2, 1]
+    while len(results) <= n:
+        results.append(sum(results[-2:]))
+    return results[n]
+
+
+def sum_series(nth, n1=0, n2=1):
+    """
+    Return the nth number in the series defined by n1 and n2
+    :param nth: number to return from within the series
+    :param n1: 1st number from which to start the series
+    :param n2: 2nd number from which to start the series
+    :return: nth number from the series
+    """
+    results = [n1, n2]
+    while len(results) <= nth:
+        results.append(sum(results[-2:]))
+    return results[nth]
+
```

Comment:
you do want to be careful about storing the whole series -- computer memory is big these days, but it's not a good habit to store a bunch of stuff you will never use.